src/servicesHost.ts

```diff
@@ -670,9 +670,19 @@ function addCache(servicesHost: typescript.ModuleResolutionHost) {
       const cache = createCache<ReturnType<typeof originalFunction>>(
         originalFunction
       );
-      servicesHost[
-        functionToCache
-      ] = cache.getCached as typescript.ModuleResolutionHost[CacheableFunction];
+      switch (functionToCache) {
```

Comment:
It's cool that you can make the compiler play ball by doing this.  However, I kind of don't like it as we're essentially writing more code to get round a problem that the compiler is presenting us with....

@andrewbranch can you think of a nicer approach than the one suggested here? (Funnily enough it looks like this compiled fine prior to TS 3.5... Something changed there I'm guessing)

Reply:
Although maybe the approach suggested here is better as it's certainly clearer... It's just the boilerplate I object to I guess.  It's one of those hard to understand compiler messages:

```
Type '((fileName: string) => boolean) | ((path: string) => string) | ((directoryName: string) => boolean) | undefined' is not assignable to type '((fileName: string) => boolean) & ((directoryName: string) => boolean) & ((path: string) => string)'.
  Type 'undefined' is not assignable to type '((fileName: string) => boolean) & ((directoryName: string) => boolean) & ((path: string) => string)'.
    Type 'undefined' is not assignable to type '(fileName: string) => boolean'.ts(2322)
```